sign/dilithium/mode/mode.go

```diff
@@ -0,0 +1,66 @@
+// mode defines the interface for the various modes of the Dilithium
+// signature scheme modes.
+package mode
+
+import (
+	"io"
+)
+
+// PublicKey is a Dilithium public key.
+type PublicKey interface {
+	// Packs public key
+	Bytes() []byte
+}
+
+// PrivateKey is a Dilithium public key.
+type PrivateKey interface {
```

Comment:
Here is where the `BinaryMarshaler` is announced (if implemented)

Reply:
The problem is that `BinaryUnmarshaler` does not work well with types hidden behind an interface.  A user is given an opaque `PrivateKey` interface type (by the key generation functions).  This is fine for marshalling, but how to use `BinaryUnmarshaler`?  The user would need to be able to create the right exact type for the mode.  So there would need to be a `CreateEmptyPrivateKey` or the like on the `Mode` interface.  That seemed less clear to me than a simple `PrivateKeyFromBytes` on the `Mode` interface.

Reply:
Is this still an issue?

Reply:
I see the problem with `BinaryUnmarshaler`, I think `PrivateKeyFromBytes` is good enough.